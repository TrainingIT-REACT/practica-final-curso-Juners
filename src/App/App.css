#main-menu {
  background-color: #e9e8e8;
  height: 100%;
  border-right: 5px solid black;
}
#main-menu .menu li > a:hover {
  background-color: #c4c4c4;
}
#main-menu .menu li > a {
  width: 100%;
  text-align: center;
  color: black;
  padding: 30px;
  font-size: 1.2rem;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
}
#main-menu .menu li > a.active {
  background-color: #c4c4c4;
}

#main-menu .menu .menu-text {
  border-bottom: 25px solid black;
  padding: 50px;
  cursor: context-menu;
}

main {
  padding: 10px;
}

main.scrollable {
  max-height: 100vh;
  overflow: auto;
}