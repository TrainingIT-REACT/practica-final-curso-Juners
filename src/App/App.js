import React from "react";
import {
  BrowserRouter as Router,
  NavLink,
  Route,
  Switch
} from "react-router-dom";
import "./App.css";
import Home from "./Pages/Home";
import AlbumsListPage from "./Pages/AlbumsListPage";

const App = () => (
  <Router>
    <div className="grid-x" style={{ height: "100%" }}>
      <nav id="main-menu" className="cell small-12 large-3">
        <ul className="vertical menu align-center">
          <li className="menu-text h2">Reactify</li>
          <li>
            <NavLink to="/" exact activeClassName="active">
              Home
            </NavLink>
          </li>
          <li>
            <NavLink to="/album" activeClassName="active">
              Albums
            </NavLink>
          </li>
          <li>
            <NavLink to="/profile" activeClassName="active">
              Profile
            </NavLink>
          </li>
        </ul>
      </nav>
      <main className="cell small-12 large-9 scrollable">
        <Switch>
          <Route path="/" exact component={Home} />
          <Route path="/album" component={AlbumsListPage} />
        </Switch>
      </main>
    </div>
  </Router>
);

export default App;
