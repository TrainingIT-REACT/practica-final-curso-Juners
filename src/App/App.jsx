import React from "react";
import { connect } from "react-redux";
import {
  BrowserRouter as Router,
  NavLink,
  Route,
  Switch
} from "react-router-dom";
import "./App.css";
import Home from "./Pages/Home";
import AlbumsListPage from "./Pages/AlbumsListPage";
import Profile from "./Pages/Profile";
import PrivateRoute from "./components/PrivateRoute";
import NotFound from "./Pages/NotFound";
import MusicPlayer from "./Pages/MusicPlayer";
import Login from "./Pages/Login";

const App = ({ user }) => (
  <React.StrictMode>
    <Router>
      <div className="grid-x" style={{ height: "100%" }}>
        <nav id="main-menu" className="cell small-12 large-3">
          <ul className="vertical menu align-center">
            <li className="menu-text h2">Reactify</li>
            <li>
              <NavLink to="/" exact activeClassName="active">
                Home
            </NavLink>
            </li>
            <li>
              <NavLink to="/album" activeClassName="active">
                Albums
            </NavLink>
            </li>
            {user.signedIn ? (
              <li>
                <NavLink to="/profile" activeClassName="active">
                  Profile
              </NavLink>
              </li>
            ) : (
                <li>
                  <NavLink to="/login" activeClassName="active">
                    Login
                </NavLink>
                </li>
              )
            }
          </ul>
        </nav>
        <main className="cell small-12 large-9 scrollable">
          <Switch>
            <Route path="/" component={Home} exact />
            <Route path="/album" component={AlbumsListPage} />
            <Route path="/player" component={MusicPlayer} exact />
            <Route path="/login" component={Login} exact />
            <PrivateRoute path="/profile" component={Profile} exact />
            <Route component={NotFound} />
          </Switch>
        </main>
      </div>
    </Router >
  </React.StrictMode>
);

const mapStateToProps = ({ user = {} }) => ({
  user
});

export default connect(mapStateToProps, {})(App);
