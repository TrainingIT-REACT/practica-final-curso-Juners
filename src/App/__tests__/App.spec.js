import React from 'react';
import { shallow } from 'enzyme';

import { App } from '../App';

describe(App, () => {
  describe("user signed in", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App user={{ signedIn: false }} />);
    })

    it('renders correctly not signed in', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("user signed in", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App user={{ signedIn: true }} />);
    })

    it('renders correctly signed in', () => {
      expect(wrapper.find("nav#main-menu .menu li").last().text()).toBe("Profile");
    });
  });
});
