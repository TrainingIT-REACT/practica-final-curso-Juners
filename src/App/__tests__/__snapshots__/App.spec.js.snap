// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App user signed in renders correctly not signed in 1`] = `
<StrictMode>
  <BrowserRouter>
    <div
      className="grid-x"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <nav
        className="cell small-12 large-3"
        id="main-menu"
      >
        <ul
          className="vertical menu align-center"
        >
          <li
            className="menu-text h2"
          >
            Reactify
          </li>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/"
            >
              Home
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              to="/album"
            >
              Albums
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              to="/login"
            >
              Login
            </NavLink>
          </li>
        </ul>
      </nav>
      <main
        className="cell small-12 large-9 scrollable"
      >
        <Switch>
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Home)",
                "type": [Function],
              }
            }
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/album"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(MusicPlayer)",
                "type": [Function],
              }
            }
            exact={true}
            path="/player"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Login)",
                "type": [Function],
              }
            }
            exact={true}
            path="/login"
          />
          <Connect(PrivateRoute)
            component={
              Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Profile)",
                "type": [Function],
              }
            }
            exact={true}
            path="/profile"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </main>
    </div>
  </BrowserRouter>
</StrictMode>
`;
